packages/react-vapor/src/components/table-hoc/TableWithNewPagination.tsx

```diff
@@ -0,0 +1,132 @@
+) => (
+    Component: React.ComponentType<ITableWithPaginationProps>
+): React.ComponentClass<ITableWithPaginationProps & React.HTMLAttributes<HTMLTableElement>> => {
+    const config = HocUtils.supplyConfig(supplier);
+    const mapStateToProps = (
+        state: IReactVaporState,
+        ownProps: ITableWithPaginationProps
+    ): ITableWithPaginationStateProps | ITableHOCOwnProps => {
+        const pageNb = NavigationSelectors.getPaginationPage(state, {id: TableHOCUtils.getPaginationId(ownProps.id)});
+
+        const perPage =
+            parseInt(
+                FlatSelectSelectors.getSelectedOptionId(state, {
+                    id: PaginationUtils.getPaginationPerPageId(ownProps.id),
+                }),
+                10
+            ) || PER_PAGE_NUMBERS[1];
+
+        const isServer = ownProps.isServer || config.isServer;
+        const length = TableSelectors.getDataCount(state, {
+            id: ownProps.id,
+            data: ownProps.data,
+            isServer,
+        });
+
+        const startingIndex = pageNb * perPage;
+        const endingIndex = startingIndex + perPage;
+
+        return {
+            pageNb,
+            perPage,
+            totalEntries: length,
+            totalPages: Math.max(Math.ceil(length / perPage), 1),
+            data: isServer ? ownProps.data : ownProps.data && sliceData(ownProps.data, startingIndex, endingIndex),
+        };
+    };
+
+    const mapDispatchToProps = (
+        dispatch: (action: IReduxAction<IReduxActionsPayload>) => void,
+        ownProps: ITableHOCOwnProps
+    ): ITableWithPaginationDispatchProps => ({
+        onMount: () => dispatch(TableWithPaginationActions.add(ownProps.id)),
+        onUnmount: () => dispatch(TableWithPaginationActions.remove(ownProps.id)),
+    });
+
+    class TableWithNewPaginationDisconnected extends React.Component<ITableHOCOwnProps & ITableWithPaginationProps> {
+        componentDidMount() {
+            this.props.onMount();
+        }
+
+        componentWillUnmount() {
+            this.props.onUnmount();
+        }
+
+        componentDidUpdate(prevProps: ITableWithPaginationProps) {
+            if (prevProps.pageNb !== this.props.pageNb || prevProps.perPage !== this.props.perPage) {
+                this.props.onUpdate?.();
+            }
+        }
+
+        render() {
+            const newProps = _.omit(this.props, [...TableWithPaginationProps]);
+            return (
+                <Component {...newProps}>
+                    <TablePagination
+                        id={this.props.id}
+                        disabled={this.props.isLoading}
+                        totalPages={this.props.totalPages}
+                        defaultPerPageSelected={
+                            config?.perPageNumbers?.length > 1
+                                ? config.perPageNumbers[1]
+                                : config.perPageNumbers[0] ?? PER_PAGE_NUMBERS[1]
```

Comment:
If the `config` of the condition has an optionnal chaining operator, you should put one there too.